src/tabletraits.jl

```diff
@@ -46,11 +47,9 @@ function table(rows::AbstractArray{T}; copy=false, kwargs...) where {T<:Union{Tu
 end
 
 function table(iter; copy=false, kw...)
-    if TableTraits.isiterable(iter)
-        table(collect_columns(IteratorInterfaceExtensions.getiterator(iter)); copy=copy, kw...)
-    elseif Tables.istable(typeof(iter))
+    try
         table(Tables.columntable(iter); copy=copy, kw...)
-    else
-        throw(ArgumentError("input satisfies neither IterableTables.jl nor Tables.jl"))
+    catch e
+        throw(ArgumentError("can't convert input to a table"))
```

Comment:
I think this is not a good solution: it entirely masks useful error messages that might be thrown in a source. For example, for all the queryverse file IO packages, one now gets this error message if one tries to open a non existing file, even though the underlying packages throw error messages that properly point out what the problem is.